subiquity/server/server.py

```diff
@@ -529,13 +532,13 @@ async def wait_for_cloudinit(self):
             self.cloud_init_ok = True
         log.debug("waited %ss for cloud-init", time.time() - ci_start)
         if "status: done" in status_txt:
-            log.debug("loading cloud config")
-            init = stages.Init()
-            init.read_cfg()
-            init.fetch(existing="trust")
-            self.cloud = init.cloudify()
-            if 'autoinstall' in self.cloud.cfg:
-                cfg = self.cloud.cfg['autoinstall']
+            log.debug(
+                "loading cloud config from %s", cloud_init_instance_data_path
+            )
+            with open(cloud_init_instance_data_path) as stream:
+                cloud_init_instance_data = json.loads(stream.read())
+            cfg = cloud_init_instance_data["merged_cfg"].get("autoinstall")
```

Comment:
In a test in a VM, I found that this was not providing autoinstall data.
I mixed in extra code to show the results of the old method and this, and the old method is finding the autoinstall data.  https://github.com/dbungert/subiquity/commit/49d249af3829dc2b8334e9f75ad1be8152c27e9d

The kvm-test script may help in reproducing this.  Create a quick config file at ~/.kvm-test.yaml
https://gist.github.com/dbungert/ecb1b053f4f0d493d5ad9b324999fc38

Then you can run:
```
./scripts/kvm-test.py --build --install -o --autoinstall
```

Reply:
@dbungert  you're absultely right. I had forgotten merged_cfg was only merged from filesystem /etc/cloud and /etc/cloud/cloud.cfg.d/ and not yet userdata as well :/. This is also something that came up in investigating this with UA-related changes which need to start looking at userdata specifically. I agree we should hold off on migration to /run/cloud-init/instance-data-sensitve.json until we can provide user-data/vendor-data content in the merged_cfg.